nodejs/week3/homework-template/src/backend/api/meals.js

```diff
@@ -0,0 +1,85 @@
+const express = require("express");
+require("express-async-errors");
+const router = express.Router();
+const knex = require("../database");
+const { sum, limit } = require("../database");
+
+//to inser a new meal
+router.post("/", async (req, res) => {
+	const newMeal = {
+		title: req.query.title,
+		description: req.query.description,
+		location: req.query.location,
+		for_when: req.query.for_when,
+		max_reservations: req.query.max_reservations,
+		price: req.query.price,
+		created_date: req.query.created_date,
+	};
+	await knex("meals").insert(newMeal);
+	res.send(`new meal inserted: ${newMeal.title}`);
+});
+
+//to get a meal by id
+router.get("/:id", async (req, res) => {
+	const mealId = parseInt(req.params.id);
+	const meal = await knex("meals").where({ id: mealId });
+	res.json(meal);
+});
+
+//to put/update a meal with specific id
+router.put("/:id", async (req, res) => {
+	let mealId = parseInt(req.params.id);
+
+	await knex("meals").where({ id: mealId }).update({ title: req.query.title });
+});
+
+router.delete("/:id", async (req, res) => {
+	let mealId = parseInt(req.params.id);
+	await knex("meals").where({ id: mealId }).delete();
+});
+
+//to get all meals from table meals and
+//Quering meals table
+router.get("/", async (req, res) => {
```

Comment:
Awesome! This is the first homework that implements this request handler correctly and takes into account all the query parameters to filter the query. Nice! ^_^